src/metabase/api/collection.clj

```diff
@@ -537,16 +538,20 @@
               ;; when fetching root collection, we might have personal collection
               (assoc row :name (collection/user->personal-collection-name (:personal_owner_id row) :user))
               (dissoc row :personal_owner_id)))]
-    (for [row rows]
-      ;; Go through this rigamarole instead of hydration because we
-      ;; don't get models back from ulterior over-query
-      ;; Previous examination with logging to DB says that there's no N+1 query for this.
-      ;; However, this was only tested on H2 and Postgres
-      (-> row
-          (assoc :can_write (mi/can-write? Collection (:id row)))
-          (dissoc :collection_position :display :moderated_status :icon
-                  :collection_preview :dataset_query :table_id :query_type :is_upload)
-          update-personal-collection))))
+      (let [visible-collection-ids (collection/permissions-set->visible-collection-ids
+                                  @api/*current-user-permissions-set*)]
+        (for [row rows]
+          ;; Go through this rigamarole instead of hydration because we
+          ;; don't get models back from ulterior over-query
+          ;; Previous examination with logging to DB says that there's no N+1 query for this.
+          ;; However, this was only tested on H2 and Postgres
```

Comment:
> this was only tested on H2 and Postgres

I might not understand what this means but I wanted to double check - is this a concern?

Reply:
Someone other than me - someone with clojure knowledge - should review the clojure code, I think.

Reply:
oh yes, good call, Dan wrote the clojure!